sdk/src/Services/S3/Custom/Transfer/Internal/UploadDirectoryCommand.cs

```diff
@@ -165,11 +164,7 @@ private void UploadProgressEventCallback(object sender, UploadProgressArgs e)
             
             var totalTransferredBytes = Interlocked.Add(ref _transferredBytes, e.IncrementTransferred - e.CompensationForRetry);
 
-            int numberOfFilesUploaded = _numberOfFilesUploaded;
-            if (e.TransferredBytes == e.TotalBytes)
-            {
-                numberOfFilesUploaded = Interlocked.Increment(ref _numberOfFilesUploaded);
-            }
+            int numberOfFilesUploaded = _numberOfFilesSuccessfullyUploaded;
```

Comment:
i think this needs to be interlocked read here to be accurate because the file uploads can complete at same time. and also can you double check we are using interlocked increment when we increase the value in the other spot